### PR TITLE
Setup Keycloak on Kubernetes

### PR DESCRIPTION
**Pull request by delvh**
Closes https://frontend.gropius.duckdns.org/components/2571e2de-a0e7-4017-b721-2fd3692ad5d6/issues/80468f7a-af56-4219-a91c-b63685e9daf4.

image

To reproduce:
- Clone repo and switch to branch
- Run `terraform init`
- Run `. test-script.sh`
- Run `terraform apply`
- Wait ~3 minutes
- The screenshot above should appear

### CODE REVIEW

File: dapr.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  namespace  = kubernetes_namespace.misarch.metadata[0].name
+
+  set {
+    name  = "auth.password"
+    value = random_password.redis.result
+  }
+}
+
+/*
+# -- comment out the two resources below when initially creating the cluster, somehow this fails to plan on the first run
```
Comment (delvh) [inline dapr.tf:22]: https://frontend.gropius.duckdns.org/components/2571e2de-a0e7-4017-b721-2fd3692ad5d6/issues/723c8fad-86b9-4d73-9ca4-9c649808f5b8

File: README.md [hunk 2]
```diff
@@ -69,10 +60,21 @@ The GraphQL Gateway, configured in `gateway.tf`, serves as the central entry poi
 4. **Initialize Terraform**: Run `terraform init` to initialize the Terraform workspace.
 5. **Apply Configuration**: Execute `terraform apply` to deploy the resources to your Kubernetes cluster.
 
+#### Auto-updating Build
```
Comment (nk-coding) [inline README.md:63]: you should mention that you still need to kill the pods if you want them to update

File: README.md [hunk 3]
```diff
@@ -29,7 +25,6 @@ All services are exposed through an Nginx ingress that is presumed to already ex
 - **main.tf**: Defines the Kubernetes namespace `misarch` and establishes image pull secrets required for pulling Docker images from external repositories.
 - **ingress.tf**: Sets up the Nginx ingress for managing external access. Configurations for SSL redirection and proxy buffer sizes are also defined here. All services to be exposed have to be configured here.
```
Comment (nk-coding) [inline README.md:26]: this does not exist currently
will it ever exist?

**Review by nk-coding — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (delvh) [reply README.md:26]: Yes, in the future.
Didn't want to remove it for now.

**Pull request merged**